src/tests/acceptance/retainStateInEchoBot.test.ts

```diff
@@ -0,0 +1,124 @@
+import * as wolf from '../..'
+import { getInitialWolfState, createStorage, TestCase, runTest, StorageLayerType } from '../helpers'
+import { Ability } from '../../types'
+
+interface UserConvoState {
+    name: string | null,
+    phrase: string | null // added "phrase" to learn more about wolf maintaining state
+}
+
+const defaultStore: UserConvoState = {
+    name: null,
+    phrase: null
+}
+
+const wolfStorage: wolf.WolfStateStorage = createStorage(getInitialWolfState())
+const convoStorage = createStorage(defaultStore)
+
+const abilities: wolf.Ability<UserConvoState, StorageLayerType<UserConvoState>>[] = [
+    {
+        name: 'greet',
+        slots: [{
+            name: 'firstName', // renamed from 'name' for clarity in testing entities 
+            query: () => 'what is your name?',
+            validate: () => ({ isValid: true, reason: null }),
+            retry: () => '',
+            onFill: () => { return; },
```

Comment:
Similar to above, `onFill` is optional and returns `void`. This can be removed as well.